src/work/Work.cpp

```diff
@@ -256,14 +256,14 @@ Work::advance()
                             << getUniqueName() << " successful, scheduling run";
         scheduleRun();
     }
-    else if (anyChildFatalFailure())
+    else if (anyChildFatalFailure() && allChildrenDone())
```

Comment:
this seems wrong: why wait for all children to be done? If you are having problems on retry it probably points to a bug in the `onReset` method (or in the destructor as the way children are reset is by just deleting them).
btw, I spotted an issue with `RunProcess` that doesn't seem to properly handle reset as `ProcessManager` doesn't even seem to have a way to shutdown processes selectively, which could be part of the problem (other problems may include things like not using temp folders properly for example).

Reply:
As discussed offline, in order to fix the problem with children not being properly cleared, we'd need to implement some sort of aborting logic. I made a note about it in the design doc, but for this iteration am going to leave it as is (as in, without waiting for children).